File: Python programming language/Lab_15/task_1.py
import math


def func_f(x):
    return math.pow(math.e, math.sin(x)) - x


def func_g(x):
    return math.log(1 + math.sqrt(x)) - math.cos(x)


def func_a_ij(i):
    return math.fabs(func_f(i) + func_g(i))


def func_n_a(m, n):
    m = min(m, n)
    max_value = float("-inf")

    for i in range(1, m+1):
        max_value = max(max_value, func_a_ij(i))

    return math.sqrt(max_value)


if __name__ == "__main__":
    print(func_n_a(1000, 1000))

File: C programming language/Lab_work4/task1.c
#include <math.h>
#include <stdio.h>

// Функция для нахождения интеграла с n шагами
double f(int n)
{
  // Начальная точка 1.0 / n, далее увеличивается на шаг 2.0 / n, при этом это
  // середина прямоугольника
  double I = 0.0;
  for (double x = 1.0 / n; x <= 2.0; x += 2.0 / n)
  {
    double y;
    if (x <= 1.0)
      y = cos(x)*exp(-pow(x, 2));
    else
      y = log(x + 1) - sqrt(4 - pow(x, 2));
    I += y;
  }
  I = (2.0 / n) * I;
  return I;
}

int main()
{
  // Запрос точности
  double e;
  printf("Введите точность --> ");
  scanf("%lf", &e);

  // Запуск цикла до нужной точности
  int n = 1;
  double In = f(n);
  double I2n = f(2*n);

  while((fabs(I2n - In) / 3) >= e)
  {
    printf("%lf для n = %d\n", I2n, n*2);
    n = n * 2;
    In = I2n;
    I2n = f(2*n);
  }

  // Вывод результата
  printf("%lf\n", I2n);

  return 0;
}
